### PR TITLE
add_device_compiler_definitions [SYCL 2020]

### PR DESCRIPTION
**Pull request by ProGTX**
Bringing https://github.com/KhronosGroup/SYCL-CTS/pull/139 into the SYCL 2020 branch.

I also noticed that `SYCL_CTS_INTEL_PI_CUDA` isn't set like it is in the 1.2.1 branch, so I included that change, I can remove it if needed.

### CODE REVIEW

**Comment by bader** [comment]
> I also noticed that `SYCL_CTS_INTEL_PI_CUDA` isn't set like it is in the 1.2.1 branch, so I included that change, I can remove it if needed.

IIRC, this macro was set because SYCL-1.2.1 used OpenCL API directly assuming that implementation is based on OpeCL API.
SYCL 2020 doesn't have such assumptions, so I think eventually we should remove all customizations under `SYCL_CTS_INTEL_PI_CUDA` macro, but I don't know if we are ready for that.
@AerialMantis, do you know if this macro is still needed for SYCL-2020 branch?

**Review by bader — APPROVED** [review]
LGTM.

**Comment by keryell** [comment]
I was just looking at this this morning comparing the SYCL 1.2.1 CTS branch and the SYCL 2020 branch and realized that there was some divergence.
Would it make sense instead to just merge the SYCL 1.2.1 branch into the SYCL 2020 branch on a regular basis and after review, to benefit from the SYCL 1.2.1 too directly?

**Review by keryell — APPROVED** [review]
Approving, even if nobody answered my question.

**Comment by bader** [comment]
> Would it make sense instead to just merge the SYCL 1.2.1 branch into the SYCL 2020 branch on a regular basis and after review, to benefit from the SYCL 1.2.1 too directly?

I think it might be difficult as SYCL 2020 is not compatible with SYCL 1.2.1, so there might be conflicts. Some SYCL 1.2.1 APIs were removed from SYCL 2020 tests and there are more changes like this pending review.
What probably make sense is to have simplified process to cherry-pick changes from SYCL 1.2.1 branch, which are compatible with SYCL 2020 and patch applies cleanly to both branches.

> IIRC, this macro was set because SYCL-1.2.1 used OpenCL API directly assuming that implementation is based on OpeCL API.
> SYCL 2020 doesn't have such assumptions, so I think eventually we should remove all customizations under `SYCL_CTS_INTEL_PI_CUDA` macro, but I don't know if we are ready for that.
> @AerialMantis, do you know if this macro is still needed for SYCL-2020 branch?

I'd like to get an answer to this question as well, but I'm okay to resolve this in follow-up commits. This should not block the merge.

**Pull request merged**
